pkg/gatherers/conditional/gather_containers_logs.go

```diff
@@ -33,52 +34,67 @@ func (g *Gatherer) BuildGatherLogsOfUnhealthyPods(paramsInterface interface{}) (
 			if err != nil {
 				return nil, []error{err}
 			}
-			return g.gatherLogsOfUnhealthyPods(ctx, kubeClient.CoreV1(), params)
+			coreClient := kubeClient.CoreV1()
+			return g.gatherContainersLogs(ctx, params, coreClient)
 		},
 		CanFail: canConditionalGathererFail,
 	}, nil
 }
 
-func (g *Gatherer) gatherLogsOfUnhealthyPods(
-	ctx context.Context, coreClient v1.CoreV1Interface, params GatherLogsOfUnhealthyPodsParams,
+func (g *Gatherer) gatherContainersLogs(
+	ctx context.Context,
+	params GatherContainersLogsParams,
+	coreClient corev1client.CoreV1Interface,
 ) ([]record.Record, []error) {
-	errs := []error{}
-	records := []record.Record{}
-
 	alertInstances, ok := g.firingAlerts[params.AlertName]
 	if !ok {
-		return nil, []error{fmt.Errorf("conditional gatherer triggered, but specified alert %q is not firing", params.AlertName)}
+		err := fmt.Errorf("conditional gather triggered, but specified alert %q is not firing", params.AlertName)
+		return nil, []error{err}
 	}
+
+	var errs []error
+	var records []record.Record
+
 	for _, alertLabels := range alertInstances {
-		alertNamespace, err := getAlertPodNamespace(alertLabels)
+		podNamespace, err := getAlertPodNamespace(alertLabels)
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
-
-		alertPod, err := getAlertPodName(alertLabels)
+		podName, err := getAlertPodName(alertLabels)
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
+		var podContainer string
+		if len(params.Container) > 0 {
+			podContainer = params.Container
+		} else {
+			podContainer, err = getAlertPodContainer(alertLabels)
+			if err != nil {
+				errs = append(errs, err)
+			}
```

Comment:
Hmm wouldn't be slightly better to call `continue` in case of error? Instead of checking https://github.com/openshift/insights-operator/pull/563/files#diff-d31fdac774b007a5e25edf343a316c64d1a2292ff5685f1e7c73fc4d63521841R85 again

Reply:
Wouldn't you gather the logs from the whole namespace in such case?

Reply:
I think that the idea is that exactly that. If we can't get the container name, go and grab it from all containers.

```// The container label may not be present for all alerts (e.g., KubePodNotReady).```

As the comment says it happens 'cause for some alerts, we don't have the container name. This implementation was done by @natiiix. Maybe he knows more details about it... but I believe that it is correct, otherwise we will have the alert being fired and not being able to collect the logs if the container name isn't present for that specific alert.

Reply:
OK you're probably right. Not from all containers but probably from `FieldSelector: fmt.Sprintf("metadata.name=%s", podName),`

Reply:
Well, with the `KubePodNotReady` status, you cannot even get a container name because the pod and its containers don't really exist yet. It may be a bit of a special case, and we can't get any logs out of it anyways in the end, but that's why I added the extra check - in case there are other alerts with only information about the namespace and pod name, but without specifying the exact container that's causing the alert. I think it makes sense for alerts like that to exist.